src/controllers/shared.controllers/director.confirm.correspondence.address.controller.ts

```diff
@@ -56,16 +56,17 @@ export const postConfirmCorrespondence = async (req: Request, res: Response, nex
       const appointmentId = officerFiling.referenceAppointmentId as string;
       const companyNumber= urlUtils.getCompanyNumberFromRequestParams(req);
       const companyAppointment: CompanyAppointment = await getCompanyAppointmentFullRecord(session, companyNumber, appointmentId);
-      if(officerFiling.isServiceAddressSameAsRegisteredOfficeAddress !== companyAppointment.serviceAddressIsSameAsRegisteredOfficeAddress){
+      if(officerFiling.isServiceAddressSameAsRegisteredOfficeAddress !== companyAppointment.serviceAddressIsSameAsRegisteredOfficeAddress ||
+        !compareAddress(officerFiling.serviceAddress,companyAppointment.serviceAddress)){
```

Comment:
Would it be possible to get into the situation where both of the sameAs flags are true but the address has been updated? Maybe by pressing back and redoing parts of the address journey after updating. If so then this would register it as a change but shouldn't.

Reply:
Maybe it should use the same logic as the residential address comparison to be safe?

Reply:
Fixed